### PR TITLE
chalice test harness

### PR DESCRIPTION
**Pull request by Bento007**
This is will be used for the testing the corpora API. ChaliceTestHarness is recycled code from [test code](https://github.com/HumanCellAtlas/fusillade/blob/b5ec9ed8fa3a45fc0f449ed01b828c2bbafc80a5/tests/infra/server.py#L9) from Fusillade.

### CODE REVIEW

File: .github/workflows/push-tests.yml [hunk 1]
```diff
@@ -41,7 +41,10 @@ jobs:
         restore-keys: |
           ${{ runner.os }}-pip-
     - name: Install dependencies
-      run: pip install -r requirements.txt
+      run: |
+        pip install -r requirements.txt
+        brew install moreutils
```
Comment (calvinnhieu) [inline .github/workflows/push-tests.yml:46]: Just curious what's this dep for?

File: tests/unit/backend/chalice/api_server/test_app.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import unittest
+
+from tests.unit.backend.chalice.api_server import BaseAPITest
+
+
+class TestAPP(BaseAPITest, unittest.TestCase):
```
Comment (calvinnhieu) [inline tests/unit/backend/chalice/api_server/test_app.py:6]: Odd capitalization or typo? `TestApp` or `TestAPI`

File: backend/chalice/api_server/Makefile [hunk 3]
```diff
@@ -14,6 +14,7 @@ export PRIVATE_SUBNET_ID_1=$(shell cut -d' ' -f1 <<<"$(PRIVATE_SUBNET_IDS)")
 export PRIVATE_SUBNET_ID_2=$(shell cut -d' ' -f2 <<<"$(PRIVATE_SUBNET_IDS)")
 export PRIVATE_SUBNET_ID_3=$(shell cut -d' ' -f3 <<<"$(PRIVATE_SUBNET_IDS)")
 export SECURITY_GROUP_ID=$(shell aws ec2 describe-security-groups | jq -rc '.SecurityGroups[] | select(.GroupName == "corpora-api-lambda-$(DEPLOYMENT_STAGE)") | .GroupId')
+export DEPLOYMENT_STAGE ?= test
```
Comment (calvinnhieu) [inline backend/chalice/api_server/Makefile:17]: Does this line (17) make 18-20 obsolete?

While this addition creates guarantees for running tests, it completely sidesteps the error checking 18-20 provides during development/developer error. If this isn't necessary (e.g. possibly added in code?), I would opt to keep the error handling intact. Thanks!

File: .github/workflows/push-tests.yml (hunk 1 above)
Comment (Bento007) [reply .github/workflows/push-tests.yml:46]: This is to run the `sponge` which is used by make package for the chalice api-server app.

File: tests/unit/backend/chalice/api_server/test_app.py (hunk 2 above)
Comment (Bento007) [reply tests/unit/backend/chalice/api_server/test_app.py:6]:
```suggestion
class TestAPI(BaseAPITest, unittest.TestCase):
```

File: backend/chalice/api_server/Makefile (hunk 3 above)
Comment (Bento007) [reply backend/chalice/api_server/Makefile]: I'll move it to .github/workflows/push-test.yml

File: tests/unit/backend/chalice/api_server/test_app.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import unittest
+
+from tests.unit.backend.chalice.api_server import BaseAPITest
+
+
+class TestAPI(BaseAPITest, unittest.TestCase):
```
Comment (maniarathi) [inline tests/unit/backend/chalice/api_server/test_app.py:6]: Curious, how long does this take? This seems more like an end2end test to me which would be more appropriate in the functional tests directory.

Comment (calvinnhieu) [reply tests/unit/backend/chalice/api_server/test_app.py:6]: +1

Comment (Bento007) [reply tests/unit/backend/chalice/api_server/test_app.py:6]: After setup, they run at millisecond. spinning up and populating the database takes longer.

**Pull request merged**